messenger.rst

```diff
@@ -900,6 +900,39 @@ serializer          How to serialize the final payload  ``Redis::SERIALIZER_PHP`
                     ``Redis::OPT_SERIALIZER`` option)
 ==================  =================================== =======
 
+.. tip::
+
+    At some point you'll likely want to scale the number of workers working on your queue.
+    Make sure you assign the correct ``consumer`` and ``group`` values in that case.
+    The more likely case is that you want every worker to work on the queue independently, reducing
+    the time needed to process the pending messages. In that case, every single worker
+    must have a different ``consumer`` option value so Redis can identify the different workers.
+    When working with Docker containers one idea might be to use the ``HOSTNAME`` environment variable:
+
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            # config/packages/messenger.yaml
+            framework:
+                messenger:
+                    transports:
+                        redis: 'redis://localhost:6379/messages/symfony/%env(HOSTNAME)%'
```

Comment:
What would we use here if we weren’t using Docker?

Reply:
You just have to make sure to pass individual consumer names. In whatever way you do that. So I can't tell you, it depends on what you use :)

Reply:
hostname ( `php -r "var_dump(gethostname());"` ) would also work in many cases for none docker setup as its unique in its network. If not because servers are not in the same network it I think he need to set it manually like the example I posted above or find another variable e.g.:

```
# Worker 1
DSN=redis://localhost:6379/messages/symfony/consumer-1
# Worker 2
DSN=redis: redis://localhost:6379/messages/symfony/consumer-2
```

Or find other case uniqueness like a domain prefix.

Reply:
E.g. for `supervisor` we need to find out how we can somehow pass on the `numprocs` as individual identifiers. I haven't tested but something like this might be working:

```
[program:messenger-consume]
command=php /path/to/your/app/bin/console messenger:consume async --time-limit=3600
user=ubuntu
numprocs=2
autostart=true
autorestart=true
process_name=%(program_name)s_%(process_num)02d
environment = PROCESS_NUM=%(process_num)d
```

And then maybe use

```
redis: 'redis://localhost:6379/messages/symfony/%env(PROCESS_NUM)%'
```

Reply:
Cool! Can we try that? To create a generic solution, we could say to use some generic env var like REDIS_CONSUMER - the difference would just be then how you *set* that, depending on your setup.

Reply:
I tried now but I'm having some issues with finding the right spot 😄 It's not anything strictly related to the Redis transport. It's for any transport that needs deterministic values so I figured I'd place it as a tip where it imho belongs: the supervisor config part 😄